### PR TITLE
Add Redis and Memcache backend  + Italian Translation

### PR DESCRIPTION
**Pull request by gablau**
Add preliminary support for Redis backend (need to test)
Add preliminary support for Memcache backend (need to test) 
Dashboard size of cache relative to current site (only Redis and Memcache backends)
Dashboard add cached pages relative to current site (only Redis and Memcache backends) 
Rename variable METHOD_MMC to METHOD_MEMCACHED to avoid misunderstandings 
Add Italian Translation + POT file

### CODE REVIEW

**Comment by Zodiac1978** [comment]
Thank you for the PR. Please be patient for a code review as we haven't so many team members who can do that. Thanks in advance!

For the Italian translation, please use GlotPress on translate.wordpress.org:
https://translate.wordpress.org/locale/it/default/wp-plugins/cachify

**Comment by timse201** [comment]
italian translation is now available via wordpress translate (96% one entry left) 
if you have installed cachify you should have noticed a translation update in your admin area

File: inc/cachify.class.php [hunk 1]
```diff
@@ -1081,18 +1136,34 @@ private static function _cache_expires()
...
+			//user redis change key: "cachify:host:scheme:path"
+			$prefix = is_ssl() ? 'https' : 'http';
+			if(empty($url)) {
+			$hash = 'cachify:'.$_SERVER['HTTP_HOST'] . ':' . $prefix .':'. $_SERVER['REQUEST_URI']; 
+			} else { 
+			$hash = 'cachify:'.parse_url($url, PHP_URL_HOST) . ':' . $prefix .':'. parse_url($url, PHP_URL_PATH) ;
+			}
+  			return $hash;
+		}
+		/* all other method */
```
Comment (swissspidy) [inline inc/cachify.class.php]: `// All other methods.`

**Comment by herrbischoff** [comment]
This is nice but must be cleaned up pretty heavily. @gablau, let me know if you need assistance. The PR is several months old and no changes have taken place since.

**Comment by swissspidy** [comment]
Agreed. It's probably easier to start with a new PR that includes multiple smaller commits that are easier to scan.

**Comment by krafit** [comment]
@gablau thank you for submitting this. As already mentioned there is still some work to do and for now I will close the PR, as I'm cleaning out our backlog a little. 
But your Memcache and Redis backend is a great idea and I'd love to work with you on this for a future release.

**Pull request closed without merging**